terraform/03_gke_cluster.tf

```diff
@@ -225,6 +224,22 @@ resource "null_resource" "delay" {
   }
 }
 
+# Rewrite ratingservice address
+resource "null_resource" "ratingservice_address_rewrite" {
+  count = var.skip_ratingservice ? 0 : 1
+
+  provisioner "local-exec" {
+    command = <<-EOT
+    kubectl wait \-\-for=condition=available \-\-timeout=600s deployment/frontend
+    kubectl set env deployment.apps/frontend RATING_SERVICE_ADDR=${one(module.ratingservice[*].service_url)}
+    kubectl delete pod -l app=frontend
+    kubectl wait \-\-for=condition=available \-\-timeout=600s deployment/frontend
```

Comment:
> The previous solution worked only because the first pod of the deployment was ready after the environment variable was updated. In result, its initialization code executed with the correct value of the env var.

The old code assumed that the pod would spin up after the environment variable was set, but there is no guarantee in the ordering. Or no guarantee that the deployment would be ready when the `set env` line was run. I was hoping to fix that to protect against those race conditions here

> The new implementation waits for the frontend deployment to complete.

Waiting for the deployment is necessary, since we are re-writing the environment variable on the deployment object. We want to make sure the deployment is in the cluster and in a known state before editing it

> The RATING_SERVICE_ADDR env var is used at initialization of the frontend service

For this reason, we then delete the old pod and wait for a fresh pod to re-initialize with the new variable and become active

Reply:
Yes, I see now. Missed the pods' restart. Given that we start only one frontend pod it should be OK to wait for the complete startup